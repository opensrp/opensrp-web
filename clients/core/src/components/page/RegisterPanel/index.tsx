import * as React from 'react';
import { Row, Col, Nav, NavItem, NavLink, TabContent, TabPane, Table } from 'reactstrap';
import './index.css';
import classnames from 'classnames';

import { ChildBasicInfo } from '../../../containers/pages/Child/Profile';

export interface RegisterPanelData {
    report: string;
    date: string;
    reporter: string;
    message: string;
}

export interface RegisterPanelProps {
    registerData: RegisterPanelData[];
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    client: any;
    tabs: string[];
}

export interface RegisterPanelState {
    activeTab: string;
}

export const defaultRegisterPanelState: RegisterPanelState = {
    activeTab: '',
};

class RegisterPanel extends React.Component<RegisterPanelProps, RegisterPanelState> {
    constructor(props: RegisterPanelProps) {
        super(props);
        this.state = defaultRegisterPanelState;
    }
    changeTab = (tabName: string) => {
        this.setState({ activeTab: tabName });
    };

    render() {
        const { client, registerData, tabs } = this.props;
        const { activeTab } = this.state;
        return (
            <>
                <Col md="12">
                    <Nav style={{ marginLeft: '2.5%' }} tabs>
                        {tabs.map((tabName: string, index: number) => {
                            return (
                                <NavItem key={index} onClick={() => this.changeTab(tabName)}>
                                    <NavLink className={classnames({ active: activeTab === tabName })}>
                                        {tabName}
                                    </NavLink>
                                </NavItem>
                            );
                        })}
                    </Nav>
                    <TabContent activeTab={'1'}>
                        <TabPane tabId="1">
                            <Row>
                                <ChildBasicInfo child={client} />
                            </Row>
                            <Row>
                                <Col className="register-section" style={{ padding: '30px' }}>
                                    <Table style={{ border: '2px solid #e8e8e9' }}>
                                        <thead style={{ backgroundColor: '#f5f5f5' }}>
                                            <tr>
                                                <td>Report</td>
                                                <td>Date</td>
                                                <td>Reporter</td>
                                                <td style={{ width: '50%' }}>Message</td>
                                            </tr>
                                        </thead>
                                        <tbody>
                                            {registerData.map((data: RegisterPanelData, index: number) => {
                                                return (
                                                    <tr key={index}>
                                                        <td>{data.report}</td>
                                                        <td> {data.date} </td>
                                                        <td>{data.reporter}</td>
                                                        <td>{data.message}</td>
                                                    </tr>
                                                );
                                            })}
                                        </tbody>
                                    </Table>
                                </Col>
                            </Row>
                        </TabPane>
                    </TabContent>
                </Col>
            </>
        );
    }
}

export default RegisterPanel;
