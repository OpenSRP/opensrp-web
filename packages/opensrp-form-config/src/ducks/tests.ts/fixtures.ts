export const fixManifestReleases = [
    {
        identifier: '1.0.11',
        json: '{"forms_version":"1.0.11","identifiers":["blood_screening.json"}',
        appId: 'org.smartregister.reveal',
        appVersion: '1.2.14',
        createdAt: '2020-06-08T15:00:38.413+02:00',
        updatedAt: '2020-06-08T15:00:38.413+02:00',
    },
    {
        identifier: '1.0.12',
        json: '{"forms_version":"1.0.12","identifiers":["blood_screening.json"}',
        appId: 'org.smartregister.reveal',
        appVersion: '1.2.14',
        createdAt: '2020-06-08T15:03:00.950+02:00',
        updatedAt: '2020-06-08T15:03:00.950+02:00',
    },
    {
        identifier: '1.0.16',
        json: '{"forms_version":"1.0.16","identifiers":["blood_screening.json"}',
        appId: 'org.smartregister.reveal',
        appVersion: '1.2.16',
        createdAt: '2020-06-09T14:40:04.189+02:00',
        updatedAt: '2020-06-09T14:40:04.189+02:00',
    },
];
