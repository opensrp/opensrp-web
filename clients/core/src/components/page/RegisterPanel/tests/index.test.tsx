import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import RegisterPanel, { RegisterPanelProps } from '..';
import * as fixtures from '../../../../store/ducks/tests/fixtures';

describe('components/RegisterPanel', () => {
    beforeEach(() => {
        jest.resetAllMocks();
    });
    it('renders without crashing', () => {
        const props: RegisterPanelProps = {
            registerData: fixtures.registerData,
            client: fixtures.client1,
            tabs: ['child-health'],
        };
        shallow(<RegisterPanel {...props} />);
    });
    it('renders the InfoCard component', () => {
        const props: RegisterPanelProps = {
            registerData: fixtures.registerData,
            client: fixtures.client1,
            tabs: ['child-health'],
        };
        const wrapper = mount(<RegisterPanel {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
