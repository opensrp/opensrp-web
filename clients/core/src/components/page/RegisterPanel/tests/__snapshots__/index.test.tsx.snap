// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RegisterPanel renders the InfoCard component 1`] = `
<RegisterPanel
  client={
    Object {
      "_id": "9b67a82d-dac7-40c0-85aa-e5976339a6b6",
      "_rev": "v1",
      "addresses": Array [
        Object {
          "addressFields": null,
          "addressType": "",
          "cityVillage": "Tha Luang",
          "country": null,
          "countyDistrict": null,
          "endDate": null,
          "geopoint": null,
          "latitude": null,
          "longitude": null,
          "postalCode": null,
          "preferred": null,
          "startDate": null,
          "stateProvince": null,
          "subDistrict": null,
          "subTown": null,
          "town": null,
        },
      ],
      "attributes": Object {
        "residence": "da765947-5e4d-49f7-9eb8-2d2d00681f65",
      },
      "baseEntityId": "71ad460c-bf76-414e-9be1-0d1b2cb1bce8",
      "birthdate": 7200000,
      "birthdateApprox": false,
      "clientApplicationVersion": 2,
      "clientDatabaseVersion": 2,
      "dateCreated": 1557670951023,
      "deathdateApprox": false,
      "firstName": "Khumpai",
      "gender": "Male",
      "identifiers": Object {
        "opensrp_id": "11096120_family",
      },
      "lastName": "Family",
      "relationships": Object {
        "family_head": Array [
          "7d97182f-d623-4553-8651-5a29d2fe3f0b",
        ],
        "primary_caregiver": Array [
          "7d97182f-d623-4553-8651-5a29d2fe3f0b",
        ],
      },
      "serverVersion": 1557670950986,
      "type": "Client",
    }
  }
  registerData={
    Array [
      Object {
        "date": "20-03-2020",
        "message": "opv1",
        "report": "0 weeks",
        "reporter": "tanvir",
      },
      Object {
        "date": "20-04-2020",
        "message": "bcg",
        "report": "4 weeks",
        "reporter": "tanvir",
      },
      Object {
        "date": "20-05-2020",
        "message": "cpv",
        "report": "6 weeks",
        "reporter": "tanvir",
      },
    ]
  }
  tabs={
    Array [
      "child-health",
    ]
  }
>
  <Col
    md="12"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <div
      className="col-md-12"
    >
      <Nav
        style={
          Object {
            "marginLeft": "2.5%",
          }
        }
        tabs={true}
        tag="ul"
        vertical={false}
      >
        <ul
          className="nav nav-tabs"
          style={
            Object {
              "marginLeft": "2.5%",
            }
          }
        >
          <NavItem
            key="0"
            onClick={[Function]}
            tag="li"
          >
            <li
              className="nav-item"
              onClick={[Function]}
            >
              <NavLink
                className=""
                tag="a"
              >
                <a
                  className="nav-link"
                  onClick={[Function]}
                >
                  child-health
                </a>
              </NavLink>
            </li>
          </NavItem>
        </ul>
      </Nav>
      <TabContent
        activeTab="1"
        tag="div"
      >
        <div
          className="tab-content"
        >
          <TabPane
            tabId="1"
            tag="div"
          >
            <div
              className="tab-pane active"
            >
              <Row
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="row"
                >
                  <ChildBasicInfo
                    child={
                      Object {
                        "_id": "9b67a82d-dac7-40c0-85aa-e5976339a6b6",
                        "_rev": "v1",
                        "addresses": Array [
                          Object {
                            "addressFields": null,
                            "addressType": "",
                            "cityVillage": "Tha Luang",
                            "country": null,
                            "countyDistrict": null,
                            "endDate": null,
                            "geopoint": null,
                            "latitude": null,
                            "longitude": null,
                            "postalCode": null,
                            "preferred": null,
                            "startDate": null,
                            "stateProvince": null,
                            "subDistrict": null,
                            "subTown": null,
                            "town": null,
                          },
                        ],
                        "attributes": Object {
                          "residence": "da765947-5e4d-49f7-9eb8-2d2d00681f65",
                        },
                        "baseEntityId": "71ad460c-bf76-414e-9be1-0d1b2cb1bce8",
                        "birthdate": 7200000,
                        "birthdateApprox": false,
                        "clientApplicationVersion": 2,
                        "clientDatabaseVersion": 2,
                        "dateCreated": 1557670951023,
                        "deathdateApprox": false,
                        "firstName": "Khumpai",
                        "gender": "Male",
                        "identifiers": Object {
                          "opensrp_id": "11096120_family",
                        },
                        "lastName": "Family",
                        "relationships": Object {
                          "family_head": Array [
                            "7d97182f-d623-4553-8651-5a29d2fe3f0b",
                          ],
                          "primary_caregiver": Array [
                            "7d97182f-d623-4553-8651-5a29d2fe3f0b",
                          ],
                        },
                        "serverVersion": 1557670950986,
                        "type": "Client",
                      }
                    }
                  >
                    <Col
                      className="info-body"
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <div
                        className="info-body col"
                      >
                        <Table
                          borderless={true}
                          className="info-table"
                          responsiveTag="div"
                          tag="table"
                        >
                          <table
                            className="info-table table table-borderless"
                          >
                            <tbody>
                              <tr>
                                <td
                                  className="info-label"
                                >
                                  HHID Number
                                </td>
                                <td>
                                  71ad460c-bf76-414e-9be1-0d1b2cb1bce8
                                </td>
                                <td
                                  className="info-label"
                                >
                                  Phone
                                </td>
                                <td />
                              </tr>
                            </tbody>
                            <tbody>
                              <tr>
                                <td
                                  className="info-label"
                                >
                                  Family Name
                                </td>
                                <td>
                                  Family
                                </td>
                                <td
                                  className="info-label"
                                >
                                  Provider
                                </td>
                                <td />
                              </tr>
                            </tbody>
                            <tbody>
                              <tr>
                                <td
                                  className="info-label"
                                >
                                  Head of Household
                                </td>
                                <td>
                                  Khumpai
                                </td>
                                <td
                                  className="info-label"
                                >
                                  Register date
                                </td>
                                <td>
                                  1557670951023
                                </td>
                              </tr>
                            </tbody>
                          </table>
                        </Table>
                      </div>
                    </Col>
                  </ChildBasicInfo>
                </div>
              </Row>
              <Row
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="row"
                >
                  <Col
                    className="register-section"
                    style={
                      Object {
                        "padding": "30px",
                      }
                    }
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    <div
                      className="register-section col"
                      style={
                        Object {
                          "padding": "30px",
                        }
                      }
                    >
                      <Table
                        responsiveTag="div"
                        style={
                          Object {
                            "border": "2px solid #e8e8e9",
                          }
                        }
                        tag="table"
                      >
                        <table
                          className="table"
                          style={
                            Object {
                              "border": "2px solid #e8e8e9",
                            }
                          }
                        >
                          <thead
                            style={
                              Object {
                                "backgroundColor": "#f5f5f5",
                              }
                            }
                          >
                            <tr>
                              <td>
                                Report
                              </td>
                              <td>
                                Date
                              </td>
                              <td>
                                Reporter
                              </td>
                              <td
                                style={
                                  Object {
                                    "width": "50%",
                                  }
                                }
                              >
                                Message
                              </td>
                            </tr>
                          </thead>
                          <tbody>
                            <tr
                              key="0"
                            >
                              <td>
                                0 weeks
                              </td>
                              <td>
                                 
                                20-03-2020
                                 
                              </td>
                              <td>
                                tanvir
                              </td>
                              <td>
                                opv1
                              </td>
                            </tr>
                            <tr
                              key="1"
                            >
                              <td>
                                4 weeks
                              </td>
                              <td>
                                 
                                20-04-2020
                                 
                              </td>
                              <td>
                                tanvir
                              </td>
                              <td>
                                bcg
                              </td>
                            </tr>
                            <tr
                              key="2"
                            >
                              <td>
                                6 weeks
                              </td>
                              <td>
                                 
                                20-05-2020
                                 
                              </td>
                              <td>
                                tanvir
                              </td>
                              <td>
                                cpv
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </Table>
                    </div>
                  </Col>
                </div>
              </Row>
            </div>
          </TabPane>
        </div>
      </TabContent>
    </div>
  </Col>
</RegisterPanel>
`;
